zk_prover/src/merkle_sum_tree/mst.rs

```diff
@@ -38,6 +39,34 @@ impl<const N_ASSETS: usize, const N_BYTES: usize> MerkleSumTree<N_ASSETS, N_BYTE
         [usize; 2 * (1 + N_ASSETS)]: Sized,
     {
         let entries = parse_csv_to_entries::<&str, N_ASSETS, N_BYTES>(path)?;
+        Self::build_tree(entries, false)
+    }
+
+    /// Builds a Merkle Sum Tree from a CSV file stored at `path`. The MST leaves are sorted by the username byte values. The CSV file must be formatted as follows:
+    ///
+    /// `username;balances`
+    ///
+    /// `dxGaEAii;11888,41163`
+    pub fn new_sorted(path: &str) -> Result<Self, Box<dyn std::error::Error>>
```

Comment:
To ensure my understanding is correct, let's assume we have a sorted MST in round 1. 
If a new user is added in round 2, would we need to reconstruct the entire MST?

Reply:
Yes.